src/style.rs

```diff
@@ -117,11 +117,10 @@ use crate::Result;
 use crate::{impl_display, Ansi, Command};
 use std::fmt;
 
-pub(crate) use self::enums::Colored;
 pub use self::{
     attributes::Attributes,
     content_style::ContentStyle,
-    enums::{Attribute, Color},
+    enums::{Attribute, Color, Colored},
```

Comment:
Colored was exposed some versions ago, but we have removed it from the API to make it easier and create one way of styling. `Colored` was used to set for and background color, since we have moved to the command API it was no longer necessary to expose it. 

> If you are happy with this exposure, then maybe it's nice to give commands for its use?

I think that the commands are not really an option here. Because they are not made for returning values, only for executing. 

If I look at the functions, you return the type `Colored` and as you said, this type is needed to distinguish the background from the foreground colors. I don't have a problem with exposing it. It is a simple structure and does not introduce any additional api complexity. 

An alternative would be to create a stand-alone function to translate an ansi code to a color. Similar things have been done with `size`, `pos`, similar to this functionality they return a value and therefore cannot be used within a command. However, we woul still need to expose the `Colored` structure. 

I do prefer to expose the Colored and keep the function in there

Reply:
> I do prefer to expose the Colored and keep the function in there

Perfect, thank you!

> I think that the commands are not really an option here. Because they are not made for returning values, only for executing.

I was more thinking a command such as `SetColored` which takes a `Colored` value and applies it, just for convenience of using these values. That would add another method to set for and background colors though.

Reply:
Right, I think it can't realy hurt to add it. I see it quite often that SetForeground and SetBackground are used together. Maybe this command would offer a nice shortcut.

Reply:
> What I originally meant was that first there were multiple ways to style. As long as the functionalities go trough the command API we are good.

That makes sense!

The obvious API is to add a command, `SetColored(pub Colored)` (or `SetColor(pub Colored)`) which set either the foreground or background color.

> I see it quite often that SetForeground and SetBackground are used together. Maybe this command would offer a nice shortcut.

The API above doesn't really provide a nice way to do that. How about an API such as:
```rust
pub struct Colors {
    pub foreground: Option<Color>,
    pub background: Option<Color>,
}

impl From<Colored> for Colors {
    // ...
}

// ...

/// A command which optionally sets the colors
pub struct SetColors(pub Colors)
``` 
That provides a shortcut to set both the foreground and background color but also means that we wouldn't need a `SetColored` command since you could write `SetColores(colored.into())`.

Reply:
Ah, thanks for pointing out, I forgot the fact that we don't want to do `SetColored(pub Colored)` because it is almost identifcal to `SetFor/BackgroundColor(Color)`. 

The proposed API seems good to me. What do you think of: `pub struct Colors(Color, Color);` were the first entry is the foreground color and the second entry is the background color?

Reply:
> What do you think of: pub struct Colors(Color, Color); were the first entry is the foreground color and the second entry is the background color?

That API also sounds good.

The main reason I suggested it with the distinct `Colors` type is to allow for conveniently applying a `Colored` value. That's probably a fairly niche use case though and it's easily done with a tiny match expression.

We could, with the first proposal, also provide a `setColors` function such as:
```rust
fn setColors(foreground: Color, background: Color) -> SetColors {
    SetColors(Colors {
        foreground: Some(foreground),
        background: Some(background),
    })
}
```

That allows you to use `setColors(fg, bg)` as a command, along with `SetColors(colored.into())` as a command, with only one actual `SetColors` command.

That makes a reasonably simple API, but not as simple as just having a `SetColors` command without a distinct type - that however doesn't make using `Colored` convenient.

I think either solution is good, maybe it's best to stick to the most simple case.

Reply:
I am not in favor of a `setColors` function, it introduces an extra way to use the crossterm API, I would rather stick with a single command. 

My question was more about something else. We can go about this in two ways:

```
pub struct Colors {
    pub foreground: Option<Color>,
    pub background: Option<Color>,
}

Colors { foreground: Green, background: Blue }
```

or 

```
pub struct Colors(pub Option<Color>, pub Option<Color>)

Colors(Gree, Blue)
```

The struct version has some overhead with the `foreground` and `background` colors and thus I tought about an alternative version.

They way doesnt matter directly, both can be used like this:

```
impl From<Colored> for Colors {
    // ...
}

// ...

/// A command which optionally sets the colors
pub struct SetColors(pub Colors)

SetColors(colored.into())
SetColors(Colored (...))
```

Maybe I do prefer your version that makes the variable name explicit.

Reply:
I've added that :)

I've also added `impl From<(Color, Color)> for Colors` which makes setting both known colors more convenient although I'm happy to remove it again if you feel like it's unnecessary.

Reply:
Sure sounds good!